include/meshio/meshio_defines.hpp

```diff
@@ -21,83 +21,93 @@ enum STLFormat {
     STL_BINARY = 1
 };
 
+/* Base class for various geometry definitions */
 template<class T>
-class Mesh {
+class GeometryData {
   public:
-    std::vector< Vec4<T> >  mPositions;
-    std::vector< Vec3<float> > mNormals;
-    std::vector< Vec2<float> > mTexcoords;
-    std::vector< Vec4<float> > mColors;
-    std::vector< unsigned > mIndices;
-
-    Mesh() {}
-
-    ~Mesh() {
+    std::vector< Vec4<T> >          mPositions;
+    std::vector< Vec3<float> >      mNormals;
+    std::vector< Vec3<T> >          mTexturePositions;
+    std::vector< Vec3<T> >          mColors;
+    std::vector< Vec3<T> >          mParameterSpaceVertices;
+    std::vector< Vec3<unsigned> >   mTriVertexIndices;
+    std::vector< Vec4<unsigned> >   mQuadVertexIndices;
+    std::vector< Vec3<unsigned> >   mTriTextureIndices;
+    std::vector< Vec4<unsigned> >   mQuadTextureIndices;
+    std::vector< Vec3<unsigned> >   mTriNormalIndices;
+    std::vector< Vec4<unsigned> >   mQuadNormalIndices;
+
+    GeometryData() {}
+
+    ~GeometryData() {
         this->clear();
     }
 
-    void resize(unsigned pSize) {
-        mPositions.resize(pSize);
-        mNormals.resize(pSize);
-        mTexcoords.resize(pSize);
-        mColors.resize(pSize);
-    }
-
-    void resizeIndices(unsigned pSize) {
-        mIndices.resize(pSize);
-    }
-
-    void clear() {
+    virtual void clear() {
         mPositions.clear();
         mNormals.clear();
-        mTexcoords.clear();
+        mTexturePositions.clear();
         mColors.clear();
-        mIndices.clear();
+        mParameterSpaceVertices.clear();
+        mTriVertexIndices.clear();
+        mQuadVertexIndices.clear();
+        mTriTextureIndices.clear();
+        mQuadTextureIndices.clear();
+        mTriNormalIndices.clear();
+        mQuadNormalIndices.clear();
+    }
+
+    virtual bool operator==(const GeometryData<T>& pGeometryDataObj) {
+        if (mPositions.size() != pGeometryDataObj.mPositions.size())
+            return false;
+        if (mNormals.size() != pGeometryDataObj.mNormals.size())
+            return false;
+        if (mTexturePositions.size()!=pGeometryDataObj.mTexturePositions.size())
+            return false;
+        if (mColors.size() != pGeometryDataObj.mColors.size())
+            return false;
+        if (mParameterSpaceVertices.size() !=
+           pGeometryDataObj.mParameterSpaceVertices.size())
+            return false;
+        if (mPositions.size())
+            for (unsigned i = 0; i < mPositions.size(); ++i)
+                if (!(mPositions[i] == pGeometryDataObj.mPositions[i]))
+                    return false;
+        if (mNormals.size())
+            for (unsigned i = 0; i < mNormals.size(); ++i)
+                if (!(mNormals[i] == pGeometryDataObj.mNormals[i]))
```

Comment:
@9prady9 This way of writing is also correct as per google coding style. As an example it is present in the code snippet just above https://google.github.io/styleguide/cppguide.html#Punctuation,_Spelling_and_Grammar

What do you suggest?